tests/testSpiderWalking.cpp

```diff
@@ -142,11 +143,73 @@ TEST(testSpiderWalking, WholeEnchilada) {
       trajectory.multiPhaseInitialValues(robot, gaussian_noise, desired_dt);
   EXPECT_LONGS_EQUAL(3847, init_vals.size());
 
+  // Compare error for all factors with expected values in file.
+  // Note, expects space after comma in csv or won't work.
+  std::string key, comm;
+  double expected;
+  std::string filename = kTestPath + std::string("/testSpiderWalking.csv");
+  std::ifstream is(filename.c_str());
+  is >> key >> expected;
+  double actual = graph.error(init_vals);
+  const double tol = 1.0;
+  EXPECT_DOUBLES_EQUAL(expected, actual, tol);
+
+  // If there is an error, create a file errors.csv with all error comparisons.
+  if (fabs(actual - expected) > tol) {
```

Comment:
This functionality is very helpful! Maybe we should make this a general function which everyone can use?

BTW, I used to write some similar functions for debugging. I hope we had this earlier.

Reply:
Different PR :-) Feel free after this lands to generalize *iff* you use it somewhere else.